### DISCUSSION TITLE
Full-text search implementation discussion

### DISCUSSION THREAD

**Discussion by gudaoxuri** [asciibook/asciibook#4] [Ideas]
The default html document generated by asciidoc is a single page, so we can easily use the browser's search function.  But asciibook generated html contains multiple pages, there is a good solution to achieve full-text search?

I thought of using https://www.algolia.com/products/search-and-discovery/hosted-search-api/ ( E.g: https://docs.taro.zone/en/docs/README ) , but this engine requires that the document must be online, it is not suitable for offline documents.

**Comment by chloerei** [comment]
Not considered before. I think online documents can use third-party services. offline documents is recommended to use epub format, epub readers generally have a good search function.

**Reply by gudaoxuri** [reply]
I think there is a very common scene.

Deploying an online document in the company's intranet, so that document can only be in html format.

But the search engine of public network environment cannot be integrated for two reasons: 
1) the internal network cannot access the public network;
2) the document may contain sensitive information.
